src/Features/CSharp/Portable/CodeRefactorings/SyncNamespace/CSharpSyncNamespaceCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,217 @@
+                {
+                    var aliasQualifier = GetAliasQualifierOpt(outerMostNode);
+                    var qualifiedNamespaceName = CreateNameSyntax(namespaceParts, aliasQualifier, namespaceParts.Length - 1);
+
+                    // We might lose some trivia associated with nodes within `outerMostNode`.  
+                    @new = SyntaxFactory.QualifiedName(qualifiedNamespaceName, nameRef).WithTriviaFrom(outerMostNode);
+                } 
+                return true;
+            }
+            else
+            {
+                old = @new = null;
+                return false;
+            } 
+        }
+
+        protected override SyntaxNode CreateUsingDirective(ImmutableArray<string> namespaceParts)
+        {
+            var name = CreateNameSyntax(namespaceParts, aliasQualifier: null, namespaceParts.Length - 1);
+            return SyntaxFactory.UsingDirective(alias: null, name: name);
+        }
+
+        protected override string EscapeIdentifier(string identifier)
+            => identifier?.EscapeIdentifier();
+
+        private static string GetAliasQualifierOpt(NameSyntax name)
+        {                          
+            while (true)
+            {
+                switch (name.Kind())
+                {
+                    case SyntaxKind.QualifiedName:
+                        name = ((QualifiedNameSyntax)name).Left;
+                        continue;
+                    case SyntaxKind.AliasQualifiedName:
+                        return ((AliasQualifiedNameSyntax)name).Alias.Identifier.ValueText;
+                }
+
+                return null;
+            }
+        }
+        
+        private NameSyntax CreateNameSyntax(ImmutableArray<string> namespaceParts, string aliasQualifier, int index)
+        {
+            var part = namespaceParts[index].EscapeIdentifier();
+            var namePiece = SyntaxFactory.IdentifierName(part);
+
+            if (index == 0)
+            {
+                return aliasQualifier == null ? (NameSyntax)namePiece : SyntaxFactory.AliasQualifiedName(aliasQualifier, namePiece);
+            }
+            else
+            {
+                return SyntaxFactory.QualifiedName(CreateNameSyntax(namespaceParts, aliasQualifier, index - 1), namePiece);
+            }                                                                              
+        }
+
+        protected override async Task<(bool, NamespaceDeclarationSyntax)> ShouldPositionTriggerRefactoringAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            
+            var namespaceDeclarationCount = root.DescendantNodes().OfType<NamespaceDeclarationSyntax>().Count();
+
+            if (namespaceDeclarationCount == 1)
+            {
+                var token = root.FindToken(position);
+                if (token.Kind() == SyntaxKind.EndOfFileToken)
```

Comment:
if you were on the EOF token how could you be intersecting the name?  seems like you could just bail here.